### PR TITLE
Remove unused internal utilities

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Follow up to #2453

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -3,8 +3,6 @@ compact <- function(x) {
   x[!is_empty]
 }
 
-"%||%" <- function(a, b) if (!is.null(a)) a else b
```
Comment (IndrajeetPatil) [inline R/utils.R:6]: `{rlang}` exports this operator, and `{rlang}` is a hard dependency of `{devtools}`, so this shouldn't be needed.

File: R/utils.R [hunk 2]
```diff
@@ -30,13 +28,6 @@ vcapply <- function(x, FUN, ...) {
   vapply(x, FUN, FUN.VALUE = character(1), ...)
 }
 
-release_bullets <- function() {
-  c(
-    '`usethis::use_latest_dependencies(TRUE, "CRAN")`',
-    NULL
-  )
-}
-
```
Comment (jennybc) [inline R/utils.R:39]: We do NOT want to remove this.

This function is consulted by `devtools::release()` and `usethis::use_release_issue()`.

Comment (IndrajeetPatil) [reply R/utils.R]: Ah, sorry I missed that `{usethis}` is accessing these internal objects from `{devtools}`.

Is this where this function is used?

https://github.com/r-lib/usethis/blob/8f045dc0a88eac82a602df56e2dd62df90971a04/R/release.R#L133-L135

At any rate, I have reverted this change.

File: R/utils.R (hunk 1 above)
Comment (jennybc) [reply R/utils.R:6]: But atm devtools does not import rlang's entire namespace, so calls to `%||%` don't actually work with this PR. It's regrettable that the tests don't reveal this, but it's true.

**Comment by jennybc** [comment]
Thanks for this but I'm not doing this sort of housekeeping on devtools atm, so I'm going to close this, since it's not a quick drive-by "merge".

**Pull request closed without merging**